ratis-logservice/src/main/java/org/apache/ratis/logservice/server/LogStateMachine.java

```diff
@@ -452,29 +503,166 @@ public void close() {
 
 
 
-  private CompletableFuture<Message> processCloseLog(LogServiceRequestProto logServiceRequestProto) {
-    CloseLogRequestProto closeLog = logServiceRequestProto.getCloseLog();
+  private CompletableFuture<Message> processChangeState(LogServiceRequestProto logServiceRequestProto) {
+    LogServiceProtos.ChangeStateLogRequestProto changeState = logServiceRequestProto.getChangeState();
     // Need to check whether the file is opened if opened close it.
     // TODO need to handle exceptions while operating with files.
+    Throwable t = verifyState(State.OPEN);
+    this.state= State.valueOf(changeState.getState().name());
```

Comment:
We should have validation on the previous state and the state being requested to be moved to. For example, `ARCHIVED` to `ARCHIVING` is invalid.